.github/workflows/changes.yml

```diff
@@ -198,181 +173,100 @@ jobs:
+      aws: ${{ steps.filter.outputs.aws }}
+      axiom: ${{ steps.filter.outputs.axiom }}
+      azure: ${{ steps.filter.outputs.azure }}
+      clickhouse: ${{ steps.filter.outputs.clickhouse }}
+      databend: ${{ steps.filter.outputs.databend }}
+      datadog: ${{ steps.filter.outputs.datadog }}
+      dnstap: ${{ steps.filter.outputs.dnstap }}
+      docker-logs: ${{ steps.filter.outputs.docker-logs }}
+      elasticsearch: ${{ steps.filter.outputs.elasticsearch }}
+      eventstoredb: ${{ steps.filter.outputs.eventstoredb }}
+      fluent: ${{ steps.filter.outputs.fluent }}
+      gcp: ${{ steps.filter.outputs.gcp }}
+      humio: ${{ steps.filter.outputs.humio }}
+      http-client: ${{ steps.filter.outputs.http-client }}
+      influxdb: ${{ steps.filter.outputs.influxdb }}
+      kafka: ${{ steps.filter.outputs.kafka }}
+      logstash: ${{ steps.filter.outputs.logstash }}
+      loki: ${{ steps.filter.outputs.loki }}
+      mongodb: ${{ steps.filter.outputs.mongodb }}
+      nats: ${{ steps.filter.outputs.nats }}
+      nginx: ${{ steps.filter.outputs.nginx }}
+      opentelemetry: ${{ steps.filter.outputs.opentelemetry }}
+      postgres: ${{ steps.filter.outputs.postgres }}
+      prometheus: ${{ steps.filter.outputs.prometheus }}
+      pulsar: ${{ steps.filter.outputs.pulsar }}
+      redis: ${{ steps.filter.outputs.redis }}
+      splunk: ${{ steps.filter.outputs.splunk }}
+      webhdfs: ${{ steps.filter.outputs.webhdfs }}
+    steps:
+      - uses: actions/checkout@v3
+
+      # creates a yaml file that contains the filters for each integration,
+      # extracted from the output of the `vdev int paths <>` command, which
+      # sources the paths from the scripts/integration/.../test.yaml file
+      - name: Create filter rules for integrations
+        run: |
+          INTEGRATIONS="amqp
+          appsignal
+          aws
+          axiom
+          azure
+          clickhouse
+          databend
+          datadog-agent
+          datadog-logs
+          datadog-metrics
+          datadog-traces
+          dnstap
+          docker-logs
+          elasticsearch
+          eventstoredb
+          fluent
+          gcp
+          humio
+          http-client
+          influxdb
+          kafka
+          logstash
+          loki
+          mongodb
+          nats
+          nginx
+          opentelemetry
+          postgres
+          prometheus
+          pulsar
+          redis
+          splunk
+          webhdfs"
+
+          echo "all-int:" > int_test_filters.yaml
+          echo '- "lib/vector-core/**"' >> int_test_filters.yaml
+
+          for integration in ${INTEGRATIONS} ; do
+            echo "${integration}:" >> int_test_filters.yaml
+            PATHS=$(cargo vdev int paths ${integration})
+            while IFS= read -r path ; do echo "- \"${path}\"" >> int_test_filters.yaml ; done <<< "$PATHS"
+          done
+
+          echo "int_test_filters.yaml="
+          cat int_test_filters.yaml
+          echo ""
```

Comment:
Why not move _all_ of this logic / scripting into `vdev`?

Reply:
🤔
Definitely do able... perhaps that should be in like a `ci` subcommand? The output of that command would be not helpful for local users of vdev. (True the file paths aren't _that_ helpful, but at least they are generic)

Reply:
It would still fit as part of `integration`, at least for now, given that the `list` subcommand works across all integrations.

Reply:
Ok, will move this into `paths` subcommand.

Reply:
addressed in [13607a5](https://github.com/vectordotdev/vector/pull/17725/commits/13607a5f75bc56a780b10d91f227bec7574af568)